File: 409-LongestPalindrome/409-LongestPalindrome.java
// Last updated: 4/21/2026, 11:25:34 PM
class Solution {
    public int longestPalindrome(String s) {
        int cnt = 0;
        HashMap<Character, Integer> map = new HashMap<>();
        for(int i = 0; i < s.length() ; i++){
            char key = s.charAt(i);
            map.put(key, map.getOrDefault(key, 0) + 1);
            if(map.get(key) % 2 == 0){
                cnt+=2;
            }
        }
        for(int i: map.values()){
            if(i % 2 == 1){
                cnt++;
                break;
            }
        }
        return cnt;
    }
}

File: 409-LongestPalindrome/409-LongestPalindrome.py
# Last updated: 4/21/2026, 11:25:29 PM
class Solution:
    def longestPalindrome(self, s: str) -> int:
        counts = Counter(s)
        odd_present = False
        cnt = 0
        for i in counts.values():
            if i % 2 :
                cnt += i - 1
                odd_present = True
            else:
                cnt += i
        if odd_present == True:
            cnt += 1
        return cnt

                 

